### ISSUE TITLE
Command: unpack module

### ISSUE THREAD

**Issue by clebs** [kyma-project/cli#1302]
**Description**

Kyma modules are released and distributed as OCI container images following the [OCM](https://github.com/gardener/component-spec) format.

The CLI needs to provide utilities to unpack the modules and be able to read its contents.

**Reasons**

Allow developers to inspect the contents of module images

**Attachments**
Implementation should be based on:
https://github.com/gardener/component-cli

**Comment by tobiscr** [comment]
@clebs : afaik we are not using CNUDIE for our Kyma modules anymore - is this still a valid task?

**Comment by clebs** [comment]
Hi @tobiscr, we are not using CNUDIE but OCM (its successor). I updated the description to make it clear.

The task is still relevant.

**Comment by Tomasz-Smelcerz-SAP** [comment]
The value of this feature is questionable now, as we have more external tools that can perform the task.
There are two tools that I tested:
- [skopeo](https://github.com/containers/skopeo)
- [oras](https://github.com/oras-project/oras)

Both tools can pull and dump the artifact into a local directory.
The `oras` tool is more promising as it is [recommended](https://www.docker.com/blog/announcing-docker-hub-oci-artifacts-support/) by the docker team for the OCI artifacts management.

The latest available `oras` version, [v1.0.0-rc.1](https://github.com/oras-project/oras/releases/tag/v1.0.0-rc.1), can do the following:

`oras copy --to-oci-layout europe-west3-docker.pkg.dev/sap-kyma-jellyfish-dev/tsm-test/foo/component-descriptors/kyma-project.io/template-operator:0.0.21 ./download`

Note: This particular command to work, you need valid credentials!
This pulls an OCI artifact into a local `./download` directory.
The specific OCI artifact is a Kyma module, pushed to the registry with `kyma alpha create module` command.
In the long URL you can recognize three parts of an OCM spec registry name:
- europe-west3-docker.pkg.dev/sap-kyma-jellyfish-dev/tsm-test/foo (repository name)
- /kyma-project.io/template-operator (module name)
- 0.0.21 (module version)

This is how the contents of `download/` directory look like after the `oras copy` completes:

```
% tree download
download
├── blobs
 │   └── sha256
 │       ├── 4418e09ce610169f40f546206ab2616f2cb23d7f619348f175bd0ff9b7afb352
 │       ├── 72fe2598b7846e37594ac83cd81450f1a668d523a5b9d4c8d4080b9519e277c7
 │       ├── c68edd870840dacb8c3d52b953afaf1d3cf6f6c253f728eb9ab0ac2545eef241
 │       ├── db86408caca4c94250d8291aa79655b84146f9cc45e0da49f05a52b3722d74a0
 │       └── f30b04b95106383c7e280217b3e98a81c6aa9e581fb4bc2fbe4403273c673ed8
├── index.json
├── ingest
└── oci-layout
```

inspecting the type of the type of files indicates their purpose:
```
% file download/blobs/sha256/*
download/blobs/sha256/4418e09ce610169f40f546206ab2616f2cb23d7f619348f175bd0ff9b7afb352: JSON data
download/blobs/sha256/72fe2598b7846e37594ac83cd81450f1a668d523a5b9d4c8d4080b9519e277c7: gzip compressed data, original size modulo 2^32 62464
download/blobs/sha256/c68edd870840dacb8c3d52b953afaf1d3cf6f6c253f728eb9ab0ac2545eef241: POSIX tar archive
download/blobs/sha256/db86408caca4c94250d8291aa79655b84146f9cc45e0da49f05a52b3722d74a0: ASCII text
download/blobs/sha256/f30b04b95106383c7e280217b3e98a81c6aa9e581fb4bc2fbe4403273c673ed8: JSON data
```

The `download/index.json` points to the OCI image manifest:

```
% cat download/index.json | jq .
{
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:f30b04b95106383c7e280217b3e98a81c6aa9e581fb4bc2fbe4403273c673ed8",
      "size": 657
    }
  ]
}
```

Reading out the manifest reveals the purpose of the layers:

```
% cat download/blobs/sha256/f30b04b95106383c7e280217b3e98a81c6aa9e581fb4bc2fbe4403273c673ed8 | jq .
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.gardener.cloud.cnudie.component.config.v1+json",
    "digest": "sha256:4418e09ce610169f40f546206ab2616f2cb23d7f619348f175bd0ff9b7afb352",
    "size": 210
  },
  "layers": [
    {
      "mediaType": "application/vnd.gardener.cloud.cnudie.component-descriptor.v2+yaml+tar",
      "digest": "sha256:c68edd870840dacb8c3d52b953afaf1d3cf6f6c253f728eb9ab0ac2545eef241",
      "size": 2560
    },
    {
      "mediaType": "application/gzip",
      "digest": "sha256:72fe2598b7846e37594ac83cd81450f1a668d523a5b9d4c8d4080b9519e277c7",
      "size": 4352
    },
    {
      "mediaType": "application/octet-stream",
      "digest": "sha256:db86408caca4c94250d8291aa79655b84146f9cc45e0da49f05a52b3722d74a0",
      "size": 250
    }
  ]
}
```

In summary: The process of "unpacking" with the oras tool requires a few manual steps, but can be easily automated with a simple bash script that needs just the `jq` tool for json parsing and some other common utlis (tar, gzip)
Considering that, I would postpone implementing that functionality in the Kyma-CLI until there's a real demand from our users to provide such a feature.

**Comment by janmedrek** [comment]
Since multiple tooling is available and the scenario is not on the basic path - we won't work on this issue.
